src/commands/Minion/hunt.ts

```diff
@@ -0,0 +1,320 @@
+			creatureName = quantity;
+			quantity = null;
+		}
+
+		const creature = Hunter.Creatures.find(creature =>
+			creature.aliases.some(
+				alias =>
+					stringMatches(alias, creatureName) ||
+					stringMatches(alias.split(' ')[0], creatureName)
+			)
+		);
+
+		if (!creature) {
+			return msg.send(
+				`That's not a valid creature to hunt. Valid creatures are ${Hunter.Creatures.map(
+					creature => creature.name
+				).join(', ')}. *for more creature info write \`${msg.cmdPrefix}hunt --creatures\`.*`
+			);
+		}
+
+		if (msg.author.skillLevel(SkillsEnum.Hunter) + (usingHuntPotion ? 2 : 0) < creature.level) {
+			return msg.send(
+				`${msg.author.minionName} needs ${creature.level} Hunter to hunt ${creature.name}.`
+			);
+		}
+
+		if (creature.qpRequired && userQP < creature.qpRequired) {
+			return msg.send(
+				`${msg.author.minionName} needs ${creature.qpRequired} Questpoints to hunt ${creature.name}.`
+			);
+		}
+
+		if (creature.prayerLvl && msg.author.skillLevel(SkillsEnum.Prayer) < creature.prayerLvl) {
+			return msg.send(
+				`${msg.author.minionName} needs ${creature.prayerLvl} Prayer to hunt ${creature.name}.`
+			);
+		}
+
+		if (creature.multiTraps) {
+			traps +=
+				Math.min(
+					Math.floor(
+						(msg.author.skillLevel(SkillsEnum.Hunter) + (usingHuntPotion ? 2 : 0)) / 20
+					),
+					5
+				) + (creature.wildy ? 1 : 0);
+		}
+
+		if (creature.itemsRequired) {
+			const requiredItems: [string, number][] = Object.entries(creature.itemsRequired);
+			for (const [itemID, qty] of requiredItems) {
+				if (!bankHasItem(userBank, parseInt(itemID), qty)) {
+					return msg.send(
+						`You don't have ${traps}x ${itemNameFromID(
+							parseInt(itemID)
+						)}, hunter tools can be bought using \`${msg.cmdPrefix}buy\`.`
+					);
+				}
+			}
+		}
+
+		let { catchTime } = creature;
+
+		// Reduce time if user is experienced hunting the creature, every hour become 1% better to a cap of 10% or 20% if tracking technique.
+		const HOUR = Time.Hour;
+		const percentReduced = Math.min(
+			Math.floor(
+				(msg.author.settings.get(UserSettings.CreatureScores)[creature.id] ?? 1) /
+					(HOUR / ((creature.catchTime * Time.Second) / traps))
+			),
+			creature.huntTechnique === 'tracking' ? 20 : 10
+		);
+		catchTime *= (100 - percentReduced) / 100;
+
+		if (percentReduced >= 1)
+			boosts.push(`${percentReduced}% for being experienced hunting this creature`);
+
+		// Reduce time by 5% if user has graceful equipped
+		if (
+			!creature.wildy &&
+			hasGracefulEquipped(msg.author.settings.get(UserSettings.Gear.Skilling))
+		) {
+			boosts.push('5% boost for using Graceful');
+			catchTime *= 0.95;
+		}
+
+		if (creature.wildy) {
+			if (!hasWildyHuntGearEquipped(msg.author.settings.get(UserSettings.Gear.Misc))) {
+				return msg.send(
+					`To hunt ${creature.name} in the wilderness you need to have Karil's leathertop/Black d'hide body and Karil's leatherskirt/Black d'hide chaps equipped in \`${msg.cmdPrefix}gear misc\`.`
```

Comment:
I did put it in misc because I don't want ppl to have to swap the mage PvM setup when they go and do hunter in the wilderness. since it's a setup only used to not get pked and also risked.

Reply:
you mean so they can have anything equipped? then I would have to balance the probability of death around that aswell in the activity.